LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift

```diff
@@ -42,20 +42,37 @@ public final class InsulinModelSelectionViewModel: ObservableObject {
 
 public struct InsulinModelSelection: View, HorizontalSizeClassOverride {
 
+    let initialInsulinModelSettings: InsulinModelSettings
     @ObservedObject var viewModel: InsulinModelSelectionViewModel
     var glucoseUnit: HKUnit
     var supportedModelSettings: SupportedInsulinModelSettings
     let mode: PresentationMode
     let appName: String
-    
+    let onSave: ((InsulinModelSettings) -> Void)?
+
     public init(
         viewModel: InsulinModelSelectionViewModel,
         glucoseUnit: HKUnit,
         supportedModelSettings: SupportedInsulinModelSettings,
         appName: String,
-        mode: PresentationMode
+        mode: PresentationMode,
+        onSave: ((InsulinModelSettings) -> Void)? = nil
     ){
-        self.viewModel = viewModel
+        initialInsulinModelSettings = viewModel.insulinModelSettings
+        if mode == .settings {
+            // Use a copy, and then write back when saving
```

Comment:
I think this is fine; the other option would be to drop the view model's existence entirely, since the selection state can be held locally (in `@State`) and persisted in `onSave`, which is how the other schedule editors work. I don't think this would be too hard, and it would simplify your special-casing in TherapySettingsViewModel, which no longer need to worry about constructing an InsulinModelSelectionViewModel.